### ISSUE TITLE
getting pretty low accuracy with SNN

### ISSUE THREAD

**Issue by mervess** [Dengyu-Wu/spkeras#1]
Hello @Dengyu-Wu,

I'm trying out _spkeras_ on LeNet-5, yet the results are not satisfactory. 
The org. ANN's accuracy is 0.9674, then it drops to 0.0980 when converted to SNN.
Can you please reason it out?

Below is the console output starting with evaluating the ANN's accuracy.
```
1875/1875 - 50s - loss: 0.2242 - accuracy: 0.9349 - val_loss: 0.0861 - val_accuracy: 0.9756
313/313 [==============================] - 5s 16ms/step - loss: 0.1085 - accuracy: 0.9674
|||||  Model is ready.  |||||
{'timesteps': 256, 'thresholding': 0.5, 'amp_factor': 100, 'signed_bit': 0, 'spike_ext': 0, 'epsilon': 0.001, 'use_bias': True, 'scaling_factor': 1, 'noneloss': False, 'method': 1}
Start Converting...
Extracting Lambda...
1/8AveragePooling2D
3/8AveragePooling2D
maximum activations: [1.0, 0.94426966, 0.99999297]
Building new model...
spikeforward_0_threshold: 100
spikeforward_1_threshold: 1.059012
spikeforward_2_threshold: 100
spikeforward_3_threshold: 1.059012
spikeforward_4_threshold: 100
spikeforward_5_threshold: 100
spikeforward_6_threshold: 100
New model generated!
|||||  Model converted to SNN.  |||||
Changing model timesteps...
New model generated!
{'timesteps': 256, 'thresholding': 0.5, 'amp_factor': 100, 'signed_bit': 0, 'spike_ext': 0, 'epsilon': 0.001, 'use_bias': True, 'scaling_factor': 1, 'noneloss': False, 'method': 1}
313/313 [==============================] - 6s 16ms/step - loss: nan - accuracy: 0.0980
|||||  Model evaluated with accuracy 0.09799999743700027.  |||||
```
I tried the code as in the usage example with different timesteps, threshold values, and scaling factors as well, but it didn't affect the output.
Note that I didn't use batch-normalization between the layers.

**Comment by Dengyu-Wu** [comment]
Hi @mervess, thanks for the message, can I ask which activation you are using for the LeNet? Spkeras can only work with ReLU at this moment.

**Comment by mervess** [comment]
No problem at all, nice work by the way.

I wasn't using _ReLU_ before, good to know. I've changed the activation func. and run again, but the accuracy remains the same.
This is the architecture I'm trying.
```
keras.models.Sequential([
    keras.layers.Conv2D(6, kernel_size=5, strides=1,  activation='relu', input_shape=train_x[0].shape, padding='same'),
    keras.layers.AveragePooling2D(), 
    keras.layers.Conv2D(16, kernel_size=5, strides=1, activation='relu', padding='valid'), 
    keras.layers.AveragePooling2D(), 
    keras.layers.Flatten(), 
    keras.layers.Dense(120, activation='relu'), 
    keras.layers.Dense(84, activation='relu'),
    keras.layers.Dense(10, activation='softmax')
])
```
Also, tried it with _SGD_ instead of _adam_ as the optimizer, no change there either.

**Comment by Dengyu-Wu** [comment]
Thanks. Spkeras cannot recognise the activation inside dense or convolutional layer. Can you use following format for the training?
 
keras.models.Sequential([
    keras.layers.Conv2D(6, kernel_size=5, strides=1,  input_shape=train_x[0].shape, padding='same'),
    keras.layers.Activation('relu'), 
    keras.layers.AveragePooling2D(), 
    keras.layers.Conv2D(16, kernel_size=5, strides=1,  padding='valid'), 
    keras.layers.Activation('relu'), 
    keras.layers.AveragePooling2D(), 
    keras.layers.Flatten(), 
    keras.layers.Dense(120), 
    keras.layers.Activation('relu'), 
    keras.layers.Dense(84),
    keras.layers.Activation('relu'), 
    keras.layers.Dense(10)
    keras.layers.Activation('softmax'), 
])

**Comment by mervess** [comment]
Works fine now!
Many thanks.
